extensions/eclipse-che-theia-plugin-remote/src/node/webviews-content-aware.ts

```diff
@@ -0,0 +1,104 @@
+/**********************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+import * as theia from '@theia/plugin';
+
+import { WebviewImpl, WebviewsExtImpl } from '@theia/plugin-ext/lib/plugin/webviews';
+
+import { Plugin } from '@theia/plugin-ext/src/common/plugin-api-rpc';
+import { Uri } from '@theia/plugin';
+import { overrideUri } from './che-content-aware-utils';
+
+export class RemoteWebview extends WebviewImpl {}
+
+export class WebviewsContentAware {
```

Comment:
Probably will need to spend some times on bringing DI in upstream for the plug-in host

Reply:
I tried to digg into this thing and from what I saw, this requires some refactoring in plugin initializing mechanism. Am I right?

Reply:
https://github.com/eclipse-theia/theia/issues/5761